tests/unit/test_charm.py

```diff
@@ -1769,47 +1771,55 @@ def test_on_get_password_secrets(harness):
         mock_event.set_results.assert_called_once_with({"password": "replication-test-password"})
 
 
-@pytest.mark.parametrize("scope", [("app"), ("unit")])
+@pytest.mark.parametrize("scope,field", [("app", "operator-password"), ("unit", "csr")])
 @patch_network_get(private_address="1.1.1.1")
-def test_get_secret_secrets(harness, scope):
+def test_get_secret_secrets(harness, scope, field):
     with (
         patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
     ):
         harness.set_leader()
 
-        assert harness.charm.get_secret(scope, "operator-password") is None
-        harness.charm.set_secret(scope, "operator-password", "test-password")
-        assert harness.charm.get_secret(scope, "operator-password") == "test-password"
+        assert harness.charm.get_secret(scope, field) is None
+        harness.charm.set_secret(scope, field, "test")
+        assert harness.charm.get_secret(scope, field) == "test"
 
 
 @patch_network_get(private_address="1.1.1.1")
-def test_set_secret(harness):
+def test_set_secret(harness, _has_secrets):
+    # after new data_interfaces version, every secret we set using set_secret
+    # should not go to the databag, but to juju secrets instead.
+    if _has_secrets:
+        return
+
     with patch("charm.PostgresqlOperatorCharm._on_leader_elected"):
         rel_id = harness.model.get_relation(PEER).id
         harness.set_leader()
 
         # Test application scope.
-        assert "password" not in harness.get_relation_data(rel_id, harness.charm.app.name)
-        harness.charm.set_secret("app", "password", "test-password")
+        assert "operator_password" not in harness.get_relation_data(rel_id, harness.charm.app.name)
```

Comment:
Is there any special reason to test with the text having an underscore instead of a hyphen? I want to know if I missed something about a specific test case (maybe testing a migration from underscore to hyphen).

Reply:
AFAIK this underscore use is needed when we want `get_secret` to return what's stored in the databag instead of the secret, as per [here](https://github.com/canonical/postgresql-operator/blob/f0fc78509468c0d14a3e7329a1c3d74f76e53b2c/src/charm.py#L257-L261). But you are right, in this specific test it was not necessary to have it. Thanks!